src/backend/distributed/transaction/transaction_management.c

```diff
@@ -535,7 +536,8 @@ CoordinatedTransactionCallback(XactEvent event, void *arg)
 			 */
 			if (!IsMainDB && MainDBConnection != NULL)
 			{
-				RunCitusMainDBQuery("COMMIT");
+				bool sendInXact = true;
+				RunCitusMainDBQuery("COMMIT", sendInXact);
```

Comment:
Should we still send COMMIT_MANAGEMENT_COMMAND_2PC / COMMIT even if didn't do RemoteTransactionBegin() in utility hook?